rust/datafusion/examples/memory_table_api.rs

```diff
@@ -51,15 +51,15 @@ fn main() -> Result<()> {
     // declare a table in memory. In spark API, this corresponds to createDataFrame(...).
     let provider = MemTable::new(schema, vec![vec![batch]])?;
     ctx.register_table("t", Box::new(provider));
-    let t = ctx.table("t")?;
+    let df = ctx.table("t")?;
```

Comment:
Shouldn't we also rename `.table` and `.register_table`?

Reply:
FWIW I think `table` is still reasonable from a DataFusion point of view as the data is creating a logical table in the context of a query, while being backed by a Dataframe. Changing to `register_dataframe` would be fine too in my opinion. I could go either way

Reply:
SparkSession has a `table` method described as `Returns the specified table/view as a DataFrame`. I tend to follow Spark conventions where possible.